crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -17,24 +18,37 @@ use tracing::info;
 pub struct C8yEndPoint {
     pub c8y_host: String,
     pub device_id: String,
-    pub c8y_internal_id: String,
+    pub token: Option<String>,
+    pub devices_internal_id: HashMap<String, String>,
 }
 
 impl C8yEndPoint {
     pub fn new(c8y_host: &str, device_id: &str, c8y_internal_id: &str) -> C8yEndPoint {
+        let mut devices_internal_id = HashMap::new();
+        if !c8y_internal_id.is_empty() {
+            devices_internal_id.insert(device_id.to_string(), c8y_internal_id.to_string());
+        }
         C8yEndPoint {
             c8y_host: c8y_host.into(),
             device_id: device_id.into(),
-            c8y_internal_id: c8y_internal_id.into(),
+            token: None,
+            devices_internal_id,
         }
     }
 
-    pub fn get_c8y_internal_id(&self) -> &str {
-        &self.c8y_internal_id
+    pub fn get_internal_id(&self, id: Option<&str>) -> Option<String> {
```

Comment:
I would keep it as `device_id`, which is more generic.

Reply:
I suggested `child_id` specifically since this API is expecting only child devices IDs as that argument and `None` for the main device. Keeping it generic would have been fine if this API accepted the main device's id as well. But this [suggestion](https://github.com/thin-edge/thin-edge.io/pull/2065#discussion_r1264984722) Didier seems like the best way to avoid all such confusion.